include/vecmath/distance.h

```diff
@@ -262,9 +267,24 @@ namespace vm {
         const auto D = a * c - b * b;
 
         if (is_zero(D, constants<T>::almost_zero())) {
-            const auto f = dot(w, v);
-            const auto z = w - f * v;
-            return line_distance<T>::Parallel(squared_length(z));
+            // parallel case
+            const T p1OnR = distance_to_projected_point(r, p1);
+            const T p2OnR = distance_to_projected_point(r, p2);
+            const T perpendicularDistSquared = squared_distance(point_at_distance(r, p1OnR), p1);
+
+            if (p1OnR < static_cast<T>(0) && p2OnR < static_cast<T>(0)) {
+                // segment completely behind ray
+                const T segmentLength = sqrt(a);
+                const auto& pointOnSegmentClosestToRay = (p1OnR > p2OnR) ? p1 : p2;
+                return line_distance<T>::Parallel(0, squared_distance(r.origin, pointOnSegmentClosestToRay), segmentLength);
```

Comment:
Shouldn't the third value be either 0 or `segmentLength`, depending on whether p1 or p2 is closer to the ray origin?

Reply:
Correct. Yeah.. this stuff is hard to get right...

Reply:
Don't I know it... make sure to add a test for this case first, since it seems we're missing coverage for this case.

Reply:
Btw, `position2` would make more sense for line segments if they weren't sorted internally. As it is, it's kind of difficult to use. It might make sense to have an ordered line segment (that's not internally sorted) and only implement `distance` on this type of line segment.

Reply:
Or we could remove the sorting altogether, it's weird even if it makes sense in the context of TB. But for a "standalone" library, it's odd. Likewise for polygons I guess.